src/metrics.cpp

```diff
@@ -121,8 +153,43 @@ int printMetrics(size_t cols, int64_t nStart, bool mining)
 
         int mined = minedBlocks.get();
         if (mined > 0) {
+            LOCK(cs_main);
+            boost::strict_lock_ptr<std::list<uint256>> u = trackedBlocks.synchronize();
+            auto consensusParams = Params().GetConsensus();
+            auto tipHeight = chainActive.Height();
+            std::string units = Params().CurrencyUnits();
+
+            // Update orphans and calculate subsidies
+            CAmount immature {0};
+            CAmount mature {0};
+            for (std::list<uint256>::iterator it = u->begin(); it != u->end(); it++) {
+                auto hash = *it;
+                if (mapBlockIndex.count(hash) > 0 &&
+                        chainActive.Contains(mapBlockIndex[hash])) {
+                    int height = mapBlockIndex[hash]->nHeight;
+                    CAmount subsidy = GetBlockSubsidy(height, consensusParams);
+                    if ((height > 0) && (height <= consensusParams.GetLastFoundersRewardBlockHeight())) {
+                        subsidy -= subsidy/5;
+                    }
+                    if (std::max(0, COINBASE_MATURITY - (tipHeight - height)) > 0) {
+                        immature += subsidy;
+                    } else {
+                        mature += subsidy;
+                    }
+                } else {
+                    it = u->erase(it);
```

Comment:
According to [the specification for `std::list::erase`](http://www.cplusplus.com/reference/list/list/erase/), this is fine.